datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+                    if data_type == "person":
+                        details_container = detail_page.find(
+                            ".//div[@id='js_visibility'][@class='col-12 col-lg-9']"
+                        )
+                        crawl_person(context, details_container, link)
+                    elif data_type == "company":
+                        details_container = detail_page.find(
+                            ".//div[@class='col-12 col-lg-9']"
+                        )
+                        crawl_company(context, details_container, link)
+                    if data_type == "vessel":
+                        details_container = detail_page.find(
+                            ".//div[@id='js_visibility']"
+                        )
+                        crawl_vessel(context, details_container, link)
+
+
+def crawl_vessel(context: Context, details_container, link):
+    data: dict[str, str] = {}
+
+    xpath_definitions = [
+        (
+            "main_info_rows",
+            ".//div[contains(@class,'tools-spec')]/div[contains(@class, 'row')]",
+        ),
+        (
+            "justification_rows",
+            ".//div[contains(@class, 'tools-frame')]/div[contains(@class, 'mb-3')]",
+        ),
+        (
+            "additional_info_rows",
+            ".//div[contains(@class, 'tools-frame')]//div[@class='mb-3' or contains(@class, 'js_visibility')]",
+        ),
+    ]
+    for name, xpath_expr in xpath_definitions:
+        rows = details_container.xpath(xpath_expr)
+        for row in rows:
+            divs = row.findall("div")
+            if len(divs) == 2:
+                label_elem, value_elem = divs
+                if "yellow" in value_elem.get("class"):
+                    data = extract_label_value_pair(label_elem, value_elem, data)
+
+    web_resources = []
+    web_links = details_container.xpath(
+        ".//div[contains(@class, 'tools-frame')]//a[contains(@class, 'long-text yellow')]"
+    )
+    for raw_link in web_links:
+        link_href = raw_link.get("href", "").strip()
+        web_resources.append(link_href)
+    data["Web resources"] = web_resources
+
+    name = data.pop("Vessel name (international according to IMO)")
+    type = data.pop("Vessel Type")
+    imo_num = data.pop("IMO")
+    description = data.pop("Category")
+    flags_former = data.pop("Flags (former)")
+
+    vessel = context.make("Vessel")
+    vessel.id = context.make_id(name, imo_num)
+    vessel.add("name", name)
+    vessel.add("imoNumber", imo_num)
+    vessel.add("type", type)
+    vessel.add("description", description)
+    vessel.add("description", data.pop("Vessel information"))
+    vessel.add("callSign", data.pop("Call sign"))
+    vessel.add("flag", data.pop("Flag (Current)"))
+    vessel.add("mmsi", data.pop("MMSI"))
+    vessel.add("buildDate", data.pop("Build year"))
+    for web_resource in data.pop("Web resources"):
+        vessel.add("sourceUrl", web_resource)
```

Comment:
@pudo am I right in thinking since these aren't our sources, we only want the detail URL we used like https://war-sanctions.gur.gov.ua/en/transport/ships/455 in sourceUrl?

Do we want to place these URLs anywhere else? Perhaps a additionalUrl prop? Or just stick to the identifying information and let people find those links at the source?

![image](https://github.com/user-attachments/assets/b679a42e-9e2c-4c57-9474-a3d3349b10cb)

Style-wise, @nvmbrasserie let's not add data to a dict just to take it from the dict a few rows later, when we could just use `web_resources` in the first place. Entity.add also accepts lists of values so no need for the loop.

Reply:
we decided to include them all in sourceUrl